src/PollSignalling.cpp

```diff
@@ -4,50 +4,59 @@
 
 #include <sys/eventfd.h>
 
-
-PollSignalling::PollSignalling()
-{
-	int pipe[2] = {FD_INVALID, FD_INVALID};
-	
 #if __APPLE__
+EventPipe::EventPipe()
+{
+	int pipe[2] = { FD_INVALID, FD_INVALID };
 	//Create pipe
-	if (::pipe(pipe)==-1)
+	if (::pipe (pipe) == -1)
 	{
 		//Error
 		throw std::runtime_error("Failed to create pipe");
 	}
-	
+
 	//Set non blocking
-	fcntl(pipe[0], F_SETFL , O_NONBLOCK);
-	fcntl(pipe[1], F_SETFL , O_NONBLOCK);
-	
-	pipeFds[0] = std::make_shared<FileDescriptor>(pipe[0]);
-	pipeFds[1] = std::make_shared<FileDescriptor>(pipe[1]);
-	
+	fcntl(pipe[0], F_SETFL, O_NONBLOCK);
+	fcntl(pipe[1], F_SETFL, O_NONBLOCK);
+
+	pipeFds[0] = pipe[0];
+	pipeFds[1] = pipe[1];
```

Comment:
missing constructor call I think
```suggestion
	pipeFds[0] = FileDescriptor(pipe[0]);
	pipeFds[1] = FileDescriptor(pipe[1]);
```

Reply:
is it really needed? not sure either

Reply:
yes, no int operation conversion overload